### ISSUE TITLE
Qt5.py

### ISSUE THREAD

**Issue by mottosso** [mottosso/Qt.py#294]
I put together a preliminary successor of Qt.py for PySide2 and PyQt5 only, an alternative being "upgrading" Qt.py itself, dropping support for Qt 4.

- [Qt5.py](https://github.com/mottosso/Qt5.py)
- [Initial discussion](https://github.com/mottosso/Qt.py/issues/293)

Benefits of a separate project

1. Co-existence. Some of your projects may still benefit from Qt 4, whereas others are Qt 5 only.
2. Clarity; users won't wonder whether *their* Qt.py supports QML or other Qt 5-only features
3. What else?

Benefits of an "upgrade"

1. Maintain userbase
2. Familiarity
3. What else?

**Comment by fredrikaverpil** [comment]
Another benefit is you can install Qt.py and Qt5.py in the same virtual environment and actually even use them in different parts of your app, depending on which parts may be running inside of a Qt4 host application or not.

**Comment by mottosso** [comment]
That's actually what I had in mind for "co-existence". I've updated the description of it to make it more clear, sorry about that!

**Comment by fredrikaverpil** [comment]
Hm. Perhaps you meant to add `Qt5.py` next to the existing `Qt.py` file in this very repo. Then we'd just keep adding new `QtN.py` files as we go (where `N` is the Qt major version).

**Comment by mottosso** [comment]
> Qt5.py which implies there could potentially be a Qt6.py down the road...

Yes, that's what I was thinking. Much like how there is PyQt4, PyQt5 (and presumably, PyQt6).

> add Qt5.py next to the existing Qt.py

That's possible, worth thinking about!

**Comment by fredrikaverpil** [comment]
>> add Qt5.py next to the existing Qt.py
>
> That's possible, worth thinking about!

We would have to figure out a way so that users can (pip) install a certain version of Qt.py and then also (pip) install a completely different version of Qt5.py though. Especially if the original Qt.py goes 2.x and starts supporting the transition between Qt5/Qt6.

I just realized, you also need to secure the project names going forward if you want new, separate repositories (such as the Qt5.py repository). If someone decides to fu** with us, they could just create Qt6.py, Qt7.py and we would have an annoying problem on our hands.

**Comment by mottosso** [comment]
> We would have to figure out a way so that users can (pip) install a certain version

That's not a problem, the `setup.py` can be called anything and we can have more than one, e.g. `setup-Qt5.py`

> I just realized, you also need to secure the project names going forward

I think they're unique under the namespace "mottosso" though.

**Comment by fredrikaverpil** [comment]
> I think they're unique under the namespace "mottosso" though.
Ah, yes. PyPi on the other hand...

**Comment by fredrikaverpil** [comment]
> That's not a problem, the setup.py can be called anything and we can have more than one, e.g. setup-Qt5.py

Oh! In that case, I'm casting my vote on this one.

**Comment by mottosso** [comment]
Same repo would however complicate issue tracking, the README and perhaps most importantly releases. There is only one stream of releases per repository, and that's how we (currently) increment the version.

**Comment by fredrikaverpil** [comment]
True. Not so great, actually.

**Comment by fredrikaverpil** [comment]
@mottosso Please see https://github.com/mottosso/Qt.py/issues/293#issuecomment-403728731
